dgp/gui/plotting/plotters.py

```diff
@@ -1,45 +1,326 @@
-# coding: utf-8
+# -*- coding: utf-8 -*-
 
 """
-Class to handle Matplotlib plotting of data to be displayed in Qt GUI
+Definitions for task specific plot interfaces.
 """
-
-from dgp.lib.etc import gen_uuid
-
 import logging
 from collections import namedtuple
-from typing import Dict, Tuple, Union
+from itertools import count
+from typing import Dict, Tuple, Union, List
 from datetime import timedelta
 
-from PyQt5.QtWidgets import QSizePolicy, QMenu, QAction, QWidget, QToolBar
-from PyQt5.QtCore import pyqtSignal, QMimeData
-from PyQt5.QtGui import QCursor, QDropEvent, QDragEnterEvent, QDragMoveEvent
+import numpy as np
+import pandas as pd
 import PyQt5.QtCore as QtCore
 import PyQt5.QtWidgets as QtWidgets
+
+from PyQt5.QtWidgets import QSizePolicy, QAction, QWidget, QMenu, QToolBar
+from PyQt5.QtCore import pyqtSignal, QMimeData
+from PyQt5.QtGui import QCursor, QDropEvent, QDragEnterEvent, QDragMoveEvent
 from matplotlib.backends.backend_qt5agg import (
-   FigureCanvasQTAgg as FigureCanvas, NavigationToolbar2QT as NavigationToolbar)
+    FigureCanvasQTAgg as FigureCanvas, NavigationToolbar2QT)
 from matplotlib.figure import Figure
-from matplotlib.axes import Axes
-from matplotlib.dates import DateFormatter, num2date, date2num
-from matplotlib.ticker import NullFormatter, NullLocator, AutoLocator
 from matplotlib.backend_bases import MouseEvent, PickEvent
 from matplotlib.patches import Rectangle
+from matplotlib.axes import Axes
+from matplotlib.dates import DateFormatter, num2date, date2num
+from matplotlib.ticker import AutoLocator
 from matplotlib.lines import Line2D
 from matplotlib.text import Annotation
-import numpy as np
 
-from dgp.lib.project import Flight
 import dgp.lib.types as types
+from dgp.lib.project import Flight
+from dgp.lib.types import DataChannel, LineUpdate
+from dgp.lib.etc import gen_uuid
+from .mplutils import *
+from .backends import BasePlot, PYQTGRAPH, MATPLOTLIB, AbstractSeriesPlotter
+from .flightregion import LinearFlightRegion
 
+import pyqtgraph as pg
+from pyqtgraph.graphicsItems.LinearRegionItem import LinearRegionItem
 
 _log = logging.getLogger(__name__)
+
+
+"""
+TODO: Many of the classes here are not used, in favor of the PyQtGraph line selection interface.
```

Comment:
I think it's ok to leave this around until we work out the remaining issues with the new plotting interface.